scripts/fast-reboot

```diff
@@ -32,6 +36,24 @@ INITRD=$(echo $KERNEL_IMAGE | sed 's/vmlinuz/initrd.img/g')
 
 sonic_asic_type=$(sonic-cfggen -y /etc/sonic/sonic_version.yml -v asic_type)
 
+# Install new FW for mellanox platforms before control plane goes down
+# So on boot switch will not spend time to upgrade FW increasing the CP downtime
+if [[ "$sonic_asic_type" == "mellanox" ]]; then
+    echo "Prepare MLNX ASIC to fast reboot: install new FW if required"
+
+    MLNX_EXIT_SUCCESS="0"
+    MLNX_EXIT_ERROR="1"
+
+    MLNX_FW_UPGRADE_SCRIPT="/usr/bin/mlnx-fw-upgrade.sh"
+
+    ${MLNX_FW_UPGRADE_SCRIPT} --upgrade
+    MLNX_EXIT_CODE="$?"
+    if [[ "${MLNX_EXIT_CODE}" != "${MLNX_EXIT_SUCCESS}" ]]; then
+        echo "Failed to burn MLNX FW: errno=${MLNX_EXIT_CODE}"
+        exit "${MLNX_EXIT_ERROR}"
```

Comment:
>exit [](start = 8, length = 4)

Why this line is different from 201811 branch? Ideally you just need cherry-pick on 201803 branch if it is easy.

Reply:
this is version of fast-reboot at point https://github.com/Azure/sonic-utilities/blob/42582872ac0285fed53141d9e4b9a1a5652bb132/scripts/fast-reboot#L53 (before warb boot changes were added)